src/tcp_cubic_search.c

```diff
@@ -191,45 +209,57 @@ static void bictcp_init(struct sock *sk)
 	struct bictcp *ca = inet_csk_ca(sk);
 
 	bictcp_reset(ca);
-
-	if (hystart)
-		bictcp_hystart_reset(sk);
-
-	if (search)
-		bictcp_search_reset(sk);
-
-	if (!hystart && initial_ssthresh)
+	
+	//////////////////////// SEARCH ////////////////////////
+	/* Reset based on the mode */
+	switch (slow_start_mode) {
+		case SS_EXIT_POINT_SEARCH:
+			bictcp_search_reset(ca);
+			break;
+		case SS_EXIT_POINT_HYSTART:
+			bictcp_hystart_reset(sk);
+			break;
+		case SS_EXIT_POINT_NONE:
+			default:
+				break;
+	}
+  	///////////////////////////////////////////////////////
+	if (slow_start_mode != SS_EXIT_POINT_HYSTART && initial_ssthresh)
```

Comment:
Line: 227. Do you really want to do this ? for this statement, the original code means when hystart is disabled and initial_ssethresh is set, we set the ssthresh for the new connection.   With current implementation,  Do we really want to set ssthresh when search is enabled and initial_ssthresh is set ?

Reply:
after discussed with Jae, we think line 229 here is good. We expect search has similar behavior as legacy slow start.